apps/dc_tools/odc/apps/dc_tools/stac_api_to_dc.py

```diff
@@ -102,27 +125,34 @@ def stac_api_to_odc(
     else:
         logging.warning("API did not return the number of items.")
 
-    # Get a generator of (stac, uri, relative_uri) tuples
-    datasets_uris = get_items(search)
-
     # Do the indexing of all the things
     success = 0
     failure = 0
 
-    for dataset, uri in datasets_uris:
-        try:
-            index_update_dataset(
-                dataset,
-                uri,
+    sys.stdout.write("\rIndexing from STAC API...\n")
+    with concurrent.futures.ThreadPoolExecutor(max_workers=50) as executor:
+        future_to_item = {
+            executor.submit(
+                process_item,
+                item,
                 dc,
                 doc2ds,
                 update_if_exists=update_if_exists,
-                allow_unsafe=allow_unsafe_changes,
-            )
-            success += 1
-        except Exception as e:
-            logging.warning(f"Failed to handle dataset: {uri} with exception {e}")
-            failure += 1
+                allow_unsafe=allow_unsafe,
+            ): item.id
+            for item in search.get_all_items()
```

Comment:
How many items might this be expected to process? If it's 'a-lot a-lot', this will chew a lot of RAM.

Reply:
`get_all_items` is lazy loaded, so it's fine :-)